src/unix/linux_like/linux/mod.rs

```diff
@@ -597,6 +624,97 @@ cfg_if! {
             }
         }
 
+        #[cfg(libc_union)]
+        #[allow(unused_unsafe)]
+        impl PartialEq for sockaddr_pppox {
+            fn eq(&self, other: &Self) -> bool {
+                self.sa_family == other.sa_family &&
+                self.sa_protocol == other.sa_protocol &&
+                match self.sa_protocol {
+                    /* PX_PROTO_OE */
+                    0 => unsafe { self.sa_addr.pppoe == other.sa_addr.pppoe },
+
+                    /* PX_PROTO_PPTP */
+                    2  => unsafe { self.sa_addr.pptp == other.sa_addr.pptp },
+
+                    _ => {
+                        debug_assert!(
+                            false,
+                            "The sockaddr_pppox only supports PX_PROTO_OE \
+                            and PX_PROTO_PPTP. sa_protocol was: {}",
+                            unsafe { self.sa_protocol.clone() }
+                        );
```

Comment:
This could be removed once it's ready to land.

Reply:
fixed